integration_test/images_test.go

```diff
@@ -0,0 +1,204 @@
+			"cpu_architecture":  "x86_64",
+			"url":               "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.8/latest/rhcos-live.x86_64.iso",
+			"rootfs_url":        "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/4.8/latest/rhcos-live-rootfs.x86_64.img",
+		},
+		{
+			"openshift_version": "pre-release",
+			"cpu_architecture":  "x86_64",
+			"url":               "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest/rhcos-live.x86_64.iso",
+			"rootfs_url":        "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest/rhcos-live-rootfs.x86_64.img",
+		},
+	}
+
+	imageDir   string
+	imageStore imagestore.ImageStore
+)
+
+var _ = Describe("Image integration tests", func() {
+	var (
+		isoFilename    string
+		imageID        string
+		assistedServer *ghttp.Server
+		imageServer    *httptest.Server
+		imageClient    *http.Client
+	)
+
+	contexts := []struct {
+		name             string
+		imageType        string
+		expectedIgnition []byte
+		expectedRamdisk  []byte
+	}{
+		{
+			name:             "full-iso",
+			imageType:        imagestore.ImageTypeFull,
+			expectedIgnition: []byte("someignitioncontent"),
+			expectedRamdisk:  nil,
+		},
+		{
+			name:             "minimal-iso-with-initrd",
+			imageType:        imagestore.ImageTypeMinimal,
+			expectedIgnition: []byte("someignitioncontent"),
+			expectedRamdisk:  []byte("someramdiskcontent"),
+		},
+		{
+			name:             "minimal-iso-without-initrd",
+			imageType:        imagestore.ImageTypeMinimal,
+			expectedIgnition: []byte("someignitioncontent"),
+			expectedRamdisk:  []byte(""),
+		},
+	}
+
+	for i := range contexts {
+		ctx := contexts[i]
+
+		Context(ctx.name, func() {
+			BeforeEach(func() {
+				imageID = uuid.New().String()
+
+				// Set up assisted service
+				assistedServer = ghttp.NewServer()
+				u, err := url.Parse(assistedServer.URL())
+				Expect(err).NotTo(HaveOccurred())
+				assistedServer.AppendHandlers(
+					ghttp.CombineHandlers(
+						ghttp.VerifyRequest("GET", fmt.Sprintf("/api/assisted-install/v2/infra-envs/%s/downloads/files", imageID), "file_name=discovery.ign"),
+						ghttp.RespondWith(http.StatusOK, ctx.expectedIgnition),
+					),
+				)
+				if ctx.expectedRamdisk != nil {
+					assistedServer.AppendHandlers(
+						ghttp.CombineHandlers(
+							ghttp.VerifyRequest("GET", fmt.Sprintf("/api/assisted-install/v2/infra-envs/%s/downloads/minimal-initrd", imageID)),
+							ghttp.RespondWith(http.StatusOK, ctx.expectedRamdisk),
+						),
+					)
+				}
+
+				// Set up image handler
+				reg := prometheus.NewRegistry()
+				handler := handlers.NewImageHandler(imageStore, reg, u.Scheme, u.Host, "", "", 1)
+				imageServer = httptest.NewServer(handler)
+				imageClient = imageServer.Client()
+			})
+
+			AfterEach(func() {
+				assistedServer.Close()
+				imageServer.Close()
+				Expect(os.Remove(isoFilename)).To(Succeed())
```

Comment:
What happens here if we failed before we write the file? Do we see the actual test failure or does it complain about this?

Reply:
Code changed:

```diff
                                        _, err = io.Copy(isoFile, resp.Body)
-                                       Expect(err).NotTo(HaveOccurred())
+                                       Expect(err).To(HaveOccurred())

```

Failure:

```
    returns a properly generated minimal-iso-without-initrd iso image [It]
    /home/dzager/go/src/github.com/openshift/assisted-image-service/integration_test/images_test.go:132

    Expected an error to have occurred.  Got:
        <nil>: nil

    /home/dzager/go/src/github.com/openshift/assisted-image-service/integration_test/images_test.go:142
------------------------------
```